test/elixir/test/basics_test.exs

```diff
@@ -58,6 +58,64 @@ defmodule BasicsTest do
     assert context[:db_name] in Couch.get("/_all_dbs").body, "Db name in _all_dbs"
   end
 
+  @tag :with_db
+  test "Database name with '+' should encode to '+'", _context do
+    set_config({"chttpd", "encode_plus", "false"})
+
+    random_number = :rand.uniform(16_000_000)
+    db_name = "random+test+db+#{random_number}"
```

Comment:
the  database is created automatically and the name can be accessed as `db_name = context[:db_name]`.

Reply:
Hmm. Never mind. I see why you are doing it this way.

Reply:
I  think you should use either the following (rely on db  creation during setup) or remove `with_db` tag from your  solution.
```
@tag :with_db_name random_db_name("random+test+db")
test "Database name with '+' should encode to '+'",  context do
    set_config({"chttpd", "encode_plus", "false"})
    db_name = context[:db_name]
    resp = Couch.get("/#{db_name}")
    assert resp.status_code == 200
    assert resp.body["db_name"] == db_name
end
```

Reply:
I have removed the `:with_db` tag from two of the tests.